base/linalg/conjarray.jl

```diff
@@ -44,18 +44,144 @@ const ConjMatrix{T,M<:AbstractMatrix} = ConjArray{T,2,M}
 IndexStyle(::CA) where {CA<:ConjArray} = IndexStyle(parent_type(CA))
 IndexStyle(::Type{CA}) where {CA<:ConjArray} = IndexStyle(parent_type(CA))
 
-@propagate_inbounds getindex(a::ConjArray{T,N}, i::Int) where {T,N} = conj(getindex(a.parent, i))
-@propagate_inbounds getindex(a::ConjArray{T,N}, i::Vararg{Int,N}) where {T,N} = conj(getindex(a.parent, i...))
+@propagate_inbounds getindex(a::ConjArray{T,N}, i::Int) where {T,N} = conj(getindex(a.parent, i))::T
+@propagate_inbounds getindex(a::ConjArray{T,N}, i::Vararg{Int,N}) where {T,N} = conj(getindex(a.parent, i...))::T
 @propagate_inbounds setindex!(a::ConjArray{T,N}, v, i::Int) where {T,N} = setindex!(a.parent, conj(v), i)
 @propagate_inbounds setindex!(a::ConjArray{T,N}, v, i::Vararg{Int,N}) where {T,N} = setindex!(a.parent, conj(v), i...)
 
 @inline similar(a::ConjArray, ::Type{T}, dims::Dims{N}) where {T,N} = similar(parent(a), T, dims)
 
 # Currently, this is default behavior for RowVector only
 @inline conj(a::ConjArray) = parent(a)
+@inline adjoint(a::ConjArray) = ConjAdjointArray(parent(a))
+
+"""
+AdjointArray(array)
+
+A lazy-view wrapper of an `AbstractArray`, taking the elementwise adjoint. This
+type is usually constructed (and unwrapped) via the [`adjoint`](@ref) function, but
+currently this is the default behavior for `RowVector` only. For other arrays, the
+`AdjointArray` constructor can be used directly.
+
+# Examples
+```jldoctest
+julia> [1+im, 1-im]'
+1×2 RowVector{Complex{Int64},AdjointArray{Complex{Int64},1,Array{Complex{Int64},1}}}:
+1-1im  1+1im
+
+julia> AdjointArray([1+im 0; 0 1-im])
+2×2 AdjointArray{Complex{Int64},2,Array{Complex{Int64},2}}:
+1-1im  0+0im
+0+0im  1+1im
+```
+"""
+struct AdjointArray{T,N,A<:AbstractArray} <: AbstractArray{T,N}
+    parent::A
+end
+
+@inline AdjointArray(a::AbstractArray{T,N}) where {T,N} = AdjointArray{adjoint_type(T),N,typeof(a)}(a)
+
+const AdjointVector{T,V<:AbstractVector} = AdjointArray{T,1,V}
```

Comment:
`const AdjointVector{T,V<:AbstractVector{T}}` unless `eltype(V)` sometimes is not `T`?